idunn/api/places.py

```diff
@@ -29,7 +30,39 @@ def validate_lang(lang):
     return lang.lower()
 
 
-def get_place(id, es: Elasticsearch, indices: IndexNames, lang=None, type=None, verbosity=DEFAULT_VERBOSITY) -> Place:
+def get_place(id, es: Elasticsearch, indices: IndexNames, headers: Headers, lang=None, type=None, verbosity=DEFAULT_VERBOSITY) -> Place:
+    custom_data = {}
+    if 'X-QwantMaps-FocusPosition' in headers:
+        pos = headers.get('X-QwantMaps-FocusPosition', '').split(';')
+        if len(pos) == 3:
+            try:
+                lon = float(pos[0])
+                lat = float(pos[1])
+                zoom = float(pos[2])
+                custom_data['lon'] = lon
+                custom_data['lat'] = lat
+                custom_data['zoom'] = zoom
+            except Exception as err:
+                logger.warning('Invalid data given through "X-QwantMaps-FocusPosition" header: ' + err)
```

Comment:
To include details about the current exception, it's cleaner to pass `exc_info=True` instead of string concatenation